containers/server-image/uyuni-configfiles-sync

```diff
@@ -0,0 +1,268 @@
+# new container image.
+#
+set -e
+
+DEFAULTS_STORAGE="/etc/uyuni-configfiles-sync/"
+SYSCONFIG_FILE="/etc/sysconfig/uyuni-configfiles-sync"
+
+help() {
+    printf "Uyuni config files defaults synchronization tool
+
+Usage:
+  uyuni-configfiles-sync [command]
+
+Available Commands:
+  init	Add persistent volume to track package and config files.
+  list	List initializated persistent volumes environments.
+  sync	Synchronize files to persistent volumes and run RPM scripts.
+"
+}
+
+init() {
+    PV=$(echo "$1/" | tr -s /)
+    if [ ! -d $DEFAULTS_STORAGE ]; then
+        mkdir $DEFAULTS_STORAGE
+    fi
+    if [ ! -f $DEFAULTS_STORAGE/.buildhash ]; then
+        echo $RANDOM | sha256sum | head -c 64 > $DEFAULTS_STORAGE/.buildhash
+    fi
+    echo -n "Initializing environment for persistent volume $PV ... "
+    TARGET="$DEFAULTS_STORAGE/$(echo $PV | sed 's/\//-/g')"
+    mkdir -p $TARGET
+    echo "$PV" > $TARGET/.pv
+    # Calculate the packages that are owning the files under this path
+    PKGNAMES=$(LANG=C rpm -q -qf $(find $PV) | egrep -v "^file .+ is not owned by any package" | sort -u)
+    # Sync the package files (non-configuration and configuration files) to our storage
+    for pkg in $PKGNAMES; do
+        for cfgfile in $(rpm -ql $pkg | grep "^$PV"); do
+            if [ -f $cfgfile ]; then
+                echo ${cfgfile#"$PV"} | rsync -a --files-from=- --exclude={'*'} $PV $TARGET
+            fi
+        done
+    done
+    echo "done!"
+}
+
+sync_file() {
+    SOURCE_FILE=$1
+    TARGET_FILE=$2
+    # Synchronize package and config files
+    if echo $LIST_CONFIG_NO_REPLACE | egrep -q "( |^)$i( |$)"; then
+        # Configuration (no replace) files
+        echo "    - $TARGET_FILE - Contains modifications made by the user. Not replacing and saving new file content as rpmnew file"
+        rsync -a $SOURCE_FILE $TARGET_FILE.rpmnew
+    else
+        # Config (replace) and rest of package files
+        echo "    - $TARGET_FILE - Contains modifications made by the user. Replacing and saving old file content as rpmsave file"
+        rsync -ab --suffix ".rpmsave" $SOURCE_FILE $TARGET_FILE
+    fi
+}
+
+sync() {
+    # We skip the sync if already triggered on this container
+    if [ ! -f $DEFAULTS_STORAGE/.buildhash ]; then
+        echo "Missing $DEFAULTS_STORAGE.buildhash file. You probably need to run 'init' command first. Skipping."
+        exit
+    elif [ -f $SYSCONFIG_FILE ] && [ "$(cat $DEFAULTS_STORAGE/.buildhash)" = "$(cat $SYSCONFIG_FILE)" ]; then
+        echo "Skipping 'sync' as it was already executed on this container"
+        exit
+    fi
+    echo "Synchronizing defaults for all initializated environments"
+    echo
+    PACKAGES_ALL=""
+    for PV_DIR in $(ls $DEFAULTS_STORAGE | grep -v "^\.buildhash$"); do
+        SOURCES=$(echo "$DEFAULTS_STORAGE/$PV_DIR/" | tr -s /)
+        PV=$(cat $SOURCES/.pv)
+        PKGNAMES=$(LANG=C rpm -q -qf $(find $PV) | egrep -v "^file .+ is not owned by any package" | sort -u)
```

Comment:
Agree, this is a limitation of the current design.

But, in case we would want to get rid of "rpm" in the future, I guess one possible solution could be to make this tool to cache whatever is needed (scriptslet, config files / flags, etc) during buildtime (where "rpm" is still available) and then make the script to consume this cached data instead of calling "rpm" during "sync" time.

Reply:
we can still add this improvement later on if required